src/som/primitives/actors/PromisePrims.java

```diff
@@ -144,17 +153,52 @@ protected final SPromise registerWhenResolved(final SPromise rcvr,
 
       SPromise  promise  = SPromise.createPromise(current, promiseResolutionBreakpoint.executeCheckIsSetAndEnabled(), false, false);
       SResolver resolver = SPromise.createResolver(promise);
+      SteppingType stepping = getStepping();
 
-      PromiseCallbackMessage pcm = new PromiseCallbackMessage(EventualMessage.getCurrentExecutingMessageId(), rcvr.getOwner(),
-          block, resolver, blockCallTarget, false, promiseResolverBreakpoint.executeCheckIsSetAndEnabled(), rcvr);
+      PromiseCallbackMessage pcm = new PromiseCallbackMessage(EventualMessage.getCurrentExecutingMessage(), EventualMessage.getCurrentExecutingMessageId(), rcvr.getOwner(),
+          block, resolver, blockCallTarget, false, promiseResolverBreakpoint.executeCheckIsSetAndEnabled(), promise, rcvr, stepping);
       registerNode.register(rcvr, pcm, current);
 
       return promise;
     }
 
+    private SteppingType getStepping() {
+      if (!VmSettings.TRUFFLE_DEBUGGER_ENABLED) {
+        return null;
+      }
+
+      SteppingStrategy strategy = ActivityThread.steppingStrategy();
+      SteppingType stepping = null;
+      boolean stop = false;
+
+      if (strategy == null) {
+        return null;
+      }
+
+      if (strategy instanceof ToMessageReceiver) {
```

Comment:
The same code is duplicated above in `validateResolverForStepping()`, why is this necessary?